openlabcmd/openlabcmd/plugins/nodepool/orphanresource.py

```diff
@@ -0,0 +1,130 @@
+import datetime
+import iso8601
+import subprocess
+
+import openstack
+
+from openlabcmd.plugins.base import Plugin
+
+# DAYs
+RESOURCE_TIMEOUT = 1
+
+
+class OrphanResourcePlugin(Plugin):
+    ptype = 'nodepool'
+    name = 'orphan_resource'
+
+    # This Plugin is only for checking, as it's a dangerous action for each.
+    def __init__(self, cloud, config):
+        super(OrphanResourcePlugin, self).__init__(cloud, config)
+        self.openstack_client = openstack.connect(cloud)
+        vm_common_white_list = [element % self.cloud
+                                for element in ['%s-openlab-zuul',
+                                                '%s-openlab-nodepool',
+                                                '%s-openlab-zookeeper']]
+        vm_addition_white_list = self.config.get(
+            'check', 'vm_white_list', fallback='').replace(' ', '').split(',')
+        self.vm_white_list = list(set(vm_common_white_list +
+                                      vm_addition_white_list))
+        self.volume_white_list = self.config.get(
+            'check', 'volume_white_list',
+            fallback='').replace(' ', '').split(',')
+
+        self.fip_white_list = self.config.get(
+            'check', 'fip_white_list', fallback='').replace(' ', '').split(',')
+
+    def _is_check_heart_beat_overtime(self, timestamp):
```

Comment:
This function is meaningless. How about "_is_overtime" or something like this? This is not related with heartbeat.